Sources/ApolloCodegenLib/ApolloCodegen.swift

```diff
@@ -305,8 +305,14 @@ public class ApolloCodegen {
     )
 
     guard graphqlErrors.isEmpty else {
-      let errorlines = graphqlErrors.flatMap({ $0.logLines })
-      CodegenLogger.log(String(describing: errorlines), logLevel: .error)
+      let errorlines = graphqlErrors.flatMap({
```

Comment:
Thinking here is by making the `sourceLocations` optional and therefore the `logLines` optional, we can determine how to display a given error, so errors with `sourceLocations` get displayed as they always have, but errors without `sourceLocations` no longer cause a precondition failure and instead just have their `name` and `message` displayed to ensure the user always gets usable error information

Reply:
This is an awesome solution!